.circleci/config.yml

```diff
@@ -1,4 +1,48 @@
-version: 2
+version: 2.1
+orbs:
+  gcp:
+    executors:
+      google-cloud:
+        docker:
+          - image: google/cloud-sdk:latest
+    commands:
+      auth:
+        parameters:
+          project_id:
+            description: Id of the GCP project.
+            type: string
+          cluster_name:
+            description: Name of the GCP cluster.
+            type: string
+          compute_zone:
+            description: Name of the GCP Compute Zone.
+            type: enum
+            enum: ["us-east4-c", "us-east4-b", "us-east4-a", "us-east1-b"]
+        steps:
+          - run:
+              name: Set up GCP
+              command: |
+                echo $GCLOUD_SERVICE_KEY > ${HOME}/gcloud-service-key.json
```

Comment:
For the future: 

The name of the Circle env var that holds a GCP service account config should reflect the name of the service account config as it's named in GCP.  As-is I'm not entirely sure *which* service account is being used here.  There is a service account called `circle-deploy`, I would need to download and compare.

I could see a scenario where we have > 1 service account Circle uses, depending on `job_type` or `project_id` this is even more relevant then.

Reply:
I get that this is a mess to unwind and redo, fine to leave it for now.

Reply:
My suspicion is if we're dealing with more than one account, we're probably edging into territory where something heavier weight like keel makes more sense. That said, one thing I considered doing was actually making the env var name a parameter---I just didn't do it here because we're consistent across our current builds in using this name.

Reply:
> My suspicion is if we're dealing with more than one account, we're probably edging into territory where something heavier weight like keel makes more sense

On that note I could see the `auth` command in a separate orb.  Let's say we're using Keel, we still lean on Circle to publish the image to a registry for us.  We're going to need `auth` in either case I think, the scope of the Service Account permissions would be different though.

We can pick that up later though.

> That said, one thing I considered doing was actually making the env var name a parameter---I just didn't do it here because we're consistent across our current builds in using this name.

Sure sure.

Reply:
Yeah, I was considering grouping all this stuff under one umbrella… We'll see what the right abstraction level is over time though.

Also note that I'm currently using the `context` to manage access to the service key env variable, which is something that's configured in Circle rather than in the orb. The context can let you control e.g. whether the value is accessible depending on whose commit triggered the build.

Reply:
>Also note that I'm currently using the context to manage access to the service key env variable, which is something that's configured in Circle rather than in the orb. The context can let you control e.g. whether the value is accessible depending on whose commit triggered the build.

Ha, was just looking at this.